source/core_mqtt.c

```diff
@@ -29,6 +29,19 @@
 
 #include "core_mqtt.h"
 #include "core_mqtt_state.h"
+
+/* MQTT_DO_NOT_USE_CUSTOM_CONFIG allows building the MQTT library
+ * without a custom config. If a custom config is provided, the
+ * MQTT_DO_NOT_USE_CUSTOM_CONFIG macro should not be defined. */
+#ifndef MQTT_DO_NOT_USE_CUSTOM_CONFIG
+/* Include custom config file before other headers. */
+    #include "core_mqtt_config.h"
+#endif
```

Comment:
The core_mqtt_config.h block should be in the defaults file to remove order dependance, and since defaults is essentially a wrapper around user config